Fix dept update error

Yang-Seungmin opened the pull request:
마이페이지 학과 관련 이슈 수정
- 학과 정보를 추가할 때 api 요청에 학번만 들어가고 학과는 들어가지 않고 있는 상태
- 학과 정보 api 요청에 추가
- user response에 학번은 존재하지만 학과가 null일 경우 dept api를 통해 학과를 추가하는 로직 추가

![screenshot_20230302_232612___720](https://user-images.githubusercontent.com/23609119/222482107-5ec2a414-99f0-4c1e-804a-7bb869de3b99.png)

Yang-Seungmin committed:
Fix dept update error

yunjaena commented on domain/src/main/java/in/koreatech/koin/domain/usecase/user/GetUserInfoUseCase.kt:
```diff
@@ -1,27 +1,41 @@
-            } catch (t: Throwable) {
-                null to userErrorHandler.handleGetUserInfoError(t)
+    suspend operator fun invoke(): Pair<User?, ErrorHandler?> {
+        if(tokenRepository.getAccessToken() == null) return null to null //익명
```
null 보다는 타입을 따로 만드는것이 좋지 않을까요?

yunjaena reviewed (commented):
리뷰 확인해주세요.

Yang-Seungmin committed:
Change user to sealed class

# Conflicts:
#	domain/src/main/java/in/koreatech/koin/domain/usecase/user/GetUserInfoUseCase.kt
#	domain/src/main/java/in/koreatech/koin/domain/usecase/user/UpdateStudentUserInfoUseCase.kt
#	koin/src/main/java/in/koreatech/koin/ui/navigation/viewmodel/KoinNavigationDrawerViewModel.kt

Yang-Seungmin committed:
Merge branch 'develop' into bugfix/fix_dept_name_error_mypage

yunjaena reviewed (approved):
LGTM
